calico_containers/pycalico/ipam.py

```diff
@@ -372,16 +490,52 @@ class HostAffinityClaimedError(DataStoreError):
     pass
 
 
+class IPAMConfigConflictError(DataStoreError):
+    """
+    Attempt to change IPAM configuration that conflict with existing
+    allocations.
+    """
+    pass
+
+
+class HostAffinityClaimedByOtherError(DataStoreError):
+    """
+    Tried to set the remove affinity of a block which has affinity claimed by
+    a different host.
+    """
+    pass
+
+
+class InvalidBlockSizeError(DataStoreError):
+    """
+    Attempting to claim or release a CIDR that is smaller than the minimum
+    block size.
+    """
+    pass
+
+
 def _block_datastore_key(block_cidr):
     """
-    Translate a block_id into a datastore key.
+    Translate a block CIDR into a datastore key.
     :param block_cidr: IPNetwork representing the block
-    :return: etcd key as string.
+    :return: etcd key as a string.
     """
     path = IPAM_BLOCK_PATH % {'version': block_cidr.version}
     return path + str(block_cidr).replace("/", "-")
 
 
+def _block_host_key(host, block_cidr):
+    """
+    Translate a block CIDR into the host block key.
```

Comment:
into the host block key... used in what?

Should mention that this is the key used to store block affinity in the host etcd tree (or something like that)